litestar/handlers/http_handlers/base.py

```diff
@@ -566,6 +570,14 @@ def on_registration(self, app: Litestar) -> None:
             self._fn = ensure_async_callable(self.fn)
             self.has_sync_callable = False
 
+        self._get_kwargs_model_for_route(route)
+
+    def _get_kwargs_model_for_route(self, route: BaseRoute) -> KwargsModel:
```

Comment:
This is needed because the same handler can have different `KwargsModel`s  for different routes, because their `path_parameters` might differ. Logically though, the `KwargsModel` belongs to the handlers, and the only thing we do with it on the routes is pass it back to the handler, which also doesn't feel too great.

This isn't the final step yet and I'm hoping to get to a point where we don't need to do this memo step here at all